src/app/component/settings-panel/settings-panel.component.ts

```diff
@@ -87,7 +92,8 @@ export class SettingsPanelComponent {
     });
     this.settingsForm.controls['speed'].valueChanges.subscribe((val) => {
       if (this.settingsForm.controls['speed'].invalid) {
-        this.settingsForm.patchValue({ speed: this.currentSpeedSetting.toString() });
+        this.settingsForm.patchValue({ speed: this.nup.formatValueAndUnit(Number(val), this.settingsService.rotationalUnit.getValue()),
```

Comment:
This will set the unit to always we be the setting's rotational unit value, ignoring the user's unit input. So 20 rps will be 20 rpm. Take a look at edit pane line 218 for a better example of how to handle this:

```
        const [success, value] = this.nup.parseLengthString(
          val!,
          this.settingsService.lengthUnit.getValue()
        );
```

You can see that the function takes the big string (with the value and unit) and converts it into a value with the units we want.

Reply:
I'm creating another function called parseRotationalLengthString that does a functionality similar to parseLengthString.